var/spack/repos/builtin/packages/py-pyside/package.py

```diff
@@ -33,17 +33,23 @@ class PyPyside(PythonPackage):
         preferred=True,
```

Comment:
Do we still want this to be preferred or does 1.2.4 work now?

Reply:
I totally overlooked that there is a preference (I was testing with specified versions :-)).
I don't know what problems there where in the past, but 1.2.4 installs fine with `--test=root` for me.
Is there anything else to test? Otherwise I remove the preference.

Reply:
I would also check `spack test run py-pyside` and try to reproduce the above comment. You can also check `ldd -r` (or `otool -L` on macOS) on any .so/.dylib libraries. If there are RPATH issues, we can always set `*LD_LIBRARY_PATH`.

Reply:
There seems to be an issue with `spack test run` because when I test a python package, I get the error
```
'str' object is not callable
```
I tested `py-pyside`. `py-pip`, `py-wheel`, `py-imortlib-metadata`

Reply:
With #38928 the tests for both versions run fine. I am a little confused because it seems that there are two sets of import tests the ones for `spack test run` and the ones for `--test=root`... (the former ran into the error, but the latter not)